src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Windows.cs

```diff
@@ -140,7 +140,11 @@ public void WaitForConnection()
 
             if (IsAsync)
             {
-                WaitForConnectionCoreAsync(CancellationToken.None).GetAwaiter().GetResult();
+                ValueTask vt = WaitForConnectionCoreAsync(CancellationToken.None);
+                if (!vt.IsCompleted)
+                {
+                    vt.AsTask().GetAwaiter().GetResult();
+                }
```

Comment:
This needs to always wait on it, in order to propagate exceptions and to mark the ValueTask consumed.  I'm actually surprised / concerned this didn't result in one of our analyzers firing (we need to investigate why it didn't flag this case).
```suggestion
                vt.AsTask().GetAwaiter().GetResult();
```